gen/out/api.jl

```diff
@@ -0,0 +1,2227 @@
+end
+
+function MPI_Win_unlock(rank, win)
+    @mpichk ccall((:MPI_Win_unlock, libmpi), Cint, (Cint, MPI_Win), rank, win)
+end
+
+function MPI_Win_unlock_all(win)
+    @mpichk ccall((:MPI_Win_unlock_all, libmpi), Cint, (MPI_Win,), win)
+end
+
+function MPI_Win_wait(win)
+    @mpichk ccall((:MPI_Win_wait, libmpi), Cint, (MPI_Win,), win)
+end
+
+function MPI_Close_port(port_name)
+    @mpichk ccall((:MPI_Close_port, libmpi), Cint, (Ptr{Cchar},), port_name)
+end
+
+function MPI_Comm_accept(port_name, info, root, comm, newcomm)
+    @mpichk ccall((:MPI_Comm_accept, libmpi), Cint, (Ptr{Cchar}, MPI_Info, Cint, MPI_Comm, Ptr{MPI_Comm}), port_name, info, root, comm, newcomm)
+end
+
+function MPI_Comm_connect(port_name, info, root, comm, newcomm)
+    @mpichk ccall((:MPI_Comm_connect, libmpi), Cint, (Ptr{Cchar}, MPI_Info, Cint, MPI_Comm, Ptr{MPI_Comm}), port_name, info, root, comm, newcomm)
+end
+
+function MPI_Comm_disconnect(comm)
+    @mpichk ccall((:MPI_Comm_disconnect, libmpi), Cint, (Ptr{MPI_Comm},), comm)
+end
+
+function MPI_Comm_get_parent(parent)
+    @mpichk ccall((:MPI_Comm_get_parent, libmpi), Cint, (Ptr{MPI_Comm},), parent)
+end
+
+function MPI_Comm_join(fd, intercomm)
+    @mpichk ccall((:MPI_Comm_join, libmpi), Cint, (Cint, Ptr{MPI_Comm}), fd, intercomm)
+end
+
+function MPI_Comm_spawn(command, argv, maxprocs, info, root, comm, intercomm, array_of_errcodes)
+    @mpichk ccall((:MPI_Comm_spawn, libmpi), Cint, (Ptr{Cchar}, Ptr{Ptr{Cchar}}, Cint, MPI_Info, Cint, MPI_Comm, Ptr{MPI_Comm}, Ptr{Cint}), command, argv, maxprocs, info, root, comm, intercomm, array_of_errcodes)
+end
+
+function MPI_Comm_spawn_multiple(count, array_of_commands, array_of_argv, array_of_maxprocs, array_of_info, root, comm, intercomm, array_of_errcodes)
+    @mpichk ccall((:MPI_Comm_spawn_multiple, libmpi), Cint, (Cint, Ptr{Ptr{Cchar}}, Ptr{Ptr{Ptr{Cchar}}}, Ptr{Cint}, Ptr{MPI_Info}, Cint, MPI_Comm, Ptr{MPI_Comm}, Ptr{Cint}), count, array_of_commands, array_of_argv, array_of_maxprocs, array_of_info, root, comm, intercomm, array_of_errcodes)
+end
+
+function MPI_Lookup_name(service_name, info, port_name)
+    @mpichk ccall((:MPI_Lookup_name, libmpi), Cint, (Ptr{Cchar}, MPI_Info, Ptr{Cchar}), service_name, info, port_name)
+end
+
+function MPI_Open_port(info, port_name)
+    @mpichk ccall((:MPI_Open_port, libmpi), Cint, (MPI_Info, Ptr{Cchar}), info, port_name)
+end
+
+function MPI_Publish_name(service_name, info, port_name)
+    @mpichk ccall((:MPI_Publish_name, libmpi), Cint, (Ptr{Cchar}, MPI_Info, Ptr{Cchar}), service_name, info, port_name)
+end
+
+function MPI_Unpublish_name(service_name, info, port_name)
+    @mpichk ccall((:MPI_Unpublish_name, libmpi), Cint, (Ptr{Cchar}, MPI_Info, Ptr{Cchar}), service_name, info, port_name)
+end
+
+function MPI_Wtick()
+    @mpichk ccall((:MPI_Wtick, libmpi), Cdouble, ())
+end
+
+function MPI_Wtime()
+    @mpichk ccall((:MPI_Wtime, libmpi), Cdouble, ())
+end
```

Comment:
These two return doubles, not error codes, so don't need the `@mpichk`

Reply:
We have a macro `@mpicall` which calls the global symbols on unix platforms (to allow for MPI profilers)
https://github.com/JuliaParallel/MPI.jl/blob/f8e4749c1ce8141458d0a88664044ec64da357ba/src/implementations.jl#L8-L27

Reply:
I've reworked the `MPIgenerator` module to add either the `@mpichk` or the `@mpicall` macro prefix.